### PR TITLE
Greg/submissionlinks

### PR DESCRIPTION
**Pull request by GodGreg**
- Added 3 more Dialogs
- Created the publish form header
- The preview link and submission links can now be accessed without login in
- Disables loading the form preview/submission if closed. preview is closed on published

Demo: https://i.gyazo.com/7474c788c6777a18391e1e7ae5b3d406.mp4

### CODE REVIEW

File: src/Components/FormCreation/Dialogs/GetFormDialog.js [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            d="M12 2C8.14 2 5 5.14 5 9C5 11.38 6.19 13.47 8 14.74V17C8 17.55 8.45 18 9 18H15C15.55 18 16 17.55 16 17V14.74C17.81 13.47 19 11.38 19 9C19 5.14 15.86 2 12 2ZM9 21C9 21.55 9.45 22 10 22H14C14.55 22 15 21.55 15 21V20H9V21ZM14 13.7L14.85 13.1C16.2 12.16 17 10.63 17 9C17 6.24 14.76 4 12 4C9.24 4 7 6.24 7 9C7 10.63 7.8 12.16 9.15 13.1L10 13.7V16H14V13.7Z"
+            fill="black"
+            fillOpacity="0.54"
+          />
+        </svg>
+        {publish ? (
+          <WarningMessageText>
+            Please review the form <b>before</b> sending to applicants Changes
+            to the form <b>after publishing</b> will require assitance from
+            Arbitrium support.
```
Comment (brittanylau) [inline src/Components/FormCreation/Dialogs/GetFormDialog.js:203]: typo: "applicants Changes" → "applicants. {line break} Changes"
typo: "assitance" → "assistance"

File: src/Components/FormCreation/PublishedFormHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  font-weight: 500;
+  font-size: 16px;
+  line-height: 24px;
+`;
+
+type Props = {
+  name: string;
+  description: string;
+  submissionLink: string;
+  onChange: (name: string, description: string) => void;
```
Comment (GodGreg) [inline src/Components/FormCreation/PublishedFormHeader.tsx:52]: Don't need this

File: src/Components/FormCreation/Submission/CreateSubmissionForm.js [hunk 3]
```diff
@@ -78,7 +89,23 @@ function CreateSubmissionForm() {
...
+      setValidLink(
+        !moment(loadForm.value.previewLink.close).isBefore(moment()) &&
+          loadForm.value.draft
+      );
+    } else {
+      const link = loadForm.value.submissionLinks.find(
+        (link) => link._id === match.params.formId
+      );
+      //Has the form closed? AND is it a draft?
+      setValidLink(
```
Comment (GodGreg) [inline src/Components/FormCreation/Submission/CreateSubmissionForm.js:104]: Should be `is not a draft`

**Comment by brittanylau** [comment]
nothing like reviewing a thousand-line PR on a sunday morning 😌

File: src/Components/FormCreation/Dialogs/ManageApplicantAccessDialog.js [hunk 4]
```diff
@@ -0,0 +1,294 @@
+import React, { useEffect } from "react";
```
Comment (GodGreg) [inline src/Components/FormCreation/Dialogs/ManageApplicantAccessDialog.js:1]: This file is not done. It is out of scope of the tickets and will be implemented with https://www.notion.so/uwblueprintexecs/Active-Epics-95f65fd1b589465e9adbe5ead3815e2e?p=0c79c8d83afd4d18bc18174c66c47d24

File: src/Components/FormCreation/Dialogs/GetFormDialog.js [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+          </WarningMessageText>
+        ) : (
+          <WarningMessageText>
+            These links are meant for reviewing the form <b>before</b> allowing
+            applicants to apply. Response submitted with this link{" "}
```
Comment (brittanylau) [inline src/Components/FormCreation/Dialogs/GetFormDialog.js:208]: typo: "Response" → "Responses"

File: src/Components/FormCreation/PublishedFormHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+            />
+            <p style={{ color: "white", fontSize: 20 }}>
+              {" "}
+              This form has been published to applicants.{" "}
+            </p>
+          </div>
+          <p style={{ color: "white" }}>
+            {" "}
+            If you published the form erroneously, make a copy of this form to
+            keep editing.{" "}
```
Comment (brittanylau) [inline src/Components/FormCreation/PublishedFormHeader.tsx:101]: Copy update: "If you published the form erroneously, make a copy of this program to keep editing."

File: src/Components/FormCreation/Dialogs/PublishedFormDialog.js [hunk 7]
```diff
@@ -189,14 +189,20 @@ function GetPreviewLinkDialog({ close, link, copyLinkToClipboard }) {
             fillOpacity="0.54"
           />
         </svg>
+
         <WarningMessageText>
-          These links are meant for reviewing the form <b>before</b> allowing
-          applicants to apply. Response submitted with this link <b>will not</b>{" "}
-          be saved.
+          <p>
+            <b>Your form has been published</b>
```
Comment (brittanylau) [inline src/Components/FormCreation/Dialogs/PublishedFormDialog.js:195]: Copy update: "Your form has been published!"

**Review by brittanylau — COMMENTED** [review]
Tested locally and things work as expected!

Reviewed the backend and reqyested a few copy changes, but will leave the frontend for someone with a bit more experience with it.

**Review by ahmedhamodi — APPROVED** [review]
LGTM.

There are a lot of inline CSS styling which I think is generally bad practice. Not sure if you want to rework it now or do it in a later PR, but thought I'd mention it.

Also it looks like most of Brittany's comments are yet to be implemented, so make sure those typos are fixed up!

File: src/Components/FormCreation/PublishedFormHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+  const copyLinkToClipboard = () => {
+    navigator.clipboard.writeText(submissionLink);
+    var dummy = document.createElement("input");
+    dummy.innerText = submissionLink;
+    document.body.appendChild(dummy);
+    dummy.setAttribute("id", "dummy_id");
+    dummy.select();
+    document.execCommand("copy");
+    dummy.remove();
+    setCopiedsubmissionLink(true);
```
[the remark this answers is not in the document]
Comment (GodGreg) [reply src/Components/FormCreation/PublishedFormHeader.tsx:77]: This is actually guy's work that I should probably be exporting instead of copying

**Pull request merged**